SIT.Manager.Avalonia/Views/MainView.axaml.cs

```diff
@@ -46,12 +32,53 @@ public partial class MainView : ReactiveUserControl<MainViewModel>
 
     }
 
-    // I hate this so much, Please if someone knows of a better way to do this make a pull request. Even microsoft docs recommend this heathenry
-    private void NavView_ItemInvoked(object? sender, NavigationViewItemInvokedEventArgs e) {
-        string requestedPage = e.InvokedItem.ToString() ?? string.Empty;
-        if (NavMenuLookup.TryGetValue(requestedPage, out Type? page)) {
-            PageNavigation pageNavigation = new(page);
+    /// <summary>
+    /// Event that before checked what button was checked comparing the name of the button, which is stupid when it comes to translating buttons.
+    /// In here we are only handing Settings Button since it's the only thing that is hardcoded here. So this is when "settings button was clicked"
+    /// </summary>
+    private void NavView_ItemInvoked(object? sender, NavigationViewItemInvokedEventArgs e)
+    {
+        if (e.IsSettingsInvoked)
+        {
+            PageNavigation pageNavigation = new(typeof(SettingsPage));
             WeakReferenceMessenger.Default.Send(new PageNavigationMessage(pageNavigation));
+            return;
         }
     }
-}
+
+    /// <summary>
+    /// <see cref="PlayButton_Clicked(object?, TappedEventArgs)"/> gets fired when you click on PlayButton in Pane View.
+    /// </summary>
+    private void PlayButton_Clicked(object? sender, TappedEventArgs e)
+    {
+        PageNavigation pageNavigation = new(typeof(PlayPage));
+        WeakReferenceMessenger.Default.Send(new PageNavigationMessage(pageNavigation));
+    }
+
+    /// <summary>
+    /// <see cref="ToolsButton_Clicked(object?, TappedEventArgs)"/> gets fired when you click on ToolsButton in Pane View.
+    /// </summary>
+    private void ToolsButton_Clicked(object? sender, TappedEventArgs e)
+    {
+        PageNavigation pageNavigation = new(typeof(ToolsPage));
+        WeakReferenceMessenger.Default.Send(new PageNavigationMessage(pageNavigation));
+    }
+
+    /// <summary>
+    /// <see cref="ServerButton_Clicked(object?, TappedEventArgs)"/> gets fired when you click on ServerButton in Pane View.
+    /// </summary>
+    private void ServerButton_Clicked(object? sender, TappedEventArgs e)
+    {
+        PageNavigation pageNavigation = new(typeof(ServerPage));
+        WeakReferenceMessenger.Default.Send(new PageNavigationMessage(pageNavigation));
+    }
+
+    /// <summary>
+    /// <see cref="ModsButton_Clicked(object?, TappedEventArgs)"/> gets fired when you click on ModsButton in Pane View.
+    /// </summary>
+    private void ModsButton_Clicked(object? sender, TappedEventArgs e)
+    {
+        PageNavigation pageNavigation = new(typeof(ModsPage));
+        WeakReferenceMessenger.Default.Send(new PageNavigationMessage(pageNavigation));
+    }
```

Comment:
I don't see how this is any better than the current system. In the end it's still hard-coded types and events you're just moving where the hard coded types lie

Reply:
> I don't see how this is any better than the current system. In the end it's still hard-coded types and events you're just moving where the hard coded types lie

Better in they way that you can switch pages that are not depended literally on string. Current system checks what current page name is. And if the page name != whatever in Dictionary we cannot go to another page. Also why to bother? They are meant to be static. I don't think we will add 10.000+ buttons in that pane what so ever.

Reply:
This is actually my fault as I appear to have accidentally swapped from using the NavigationViewItem's Tag property to the InvokedItem which gives the display name. The Tag property allows us to have an identifier separate from the display value without needing an event for each item. I will push the needed changes for this now